airflow/providers/amazon/aws/hooks/neptune.py

```diff
@@ -0,0 +1,113 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Interact with AWS Neptune."""
+from __future__ import annotations
+
+import time
+from typing import Callable
+
+from airflow.exceptions import AirflowException, AirflowNotFoundException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class NeptuneHook(AwsBaseHook):
+    """
+    Interact with AWS Neptune using proper client from the boto3 library.
+
+    Hook attribute `conn` has all methods that listed in documentation
+
+    .. seealso::
+        - https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/neptune.html
+        - https://docs.aws.amazon.com/neptune/index.html
+
+    Additional arguments (such as ``aws_conn_id`` or ``region_name``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    .. seealso::
+        :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsGenericHook`
+
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        kwargs["client_type"] = "neptune"
+        super().__init__(*args, **kwargs)
+
+    def get_db_cluster_state(self, db_cluster_id: str) -> str:
+        """
+        Get the current state of a DB cluster.
+
+        :param db_cluster_id: The ID of the target DB cluster.
+        :return: Returns the status of the DB cluster as a string (eg. "available")
+        :rtype: str
+        :raises AirflowNotFoundException: If the DB cluster does not exist.
+        """
+        try:
+            response = self.conn.describe_db_clusters(DBClusterIdentifier=db_cluster_id)
+        except self.conn.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "DBClusterNotFoundFault":
+                raise AirflowNotFoundException(e)
+            raise e
+        return response["DBClusters"][0]["Status"].lower()
+
+    def wait_for_db_cluster_state(
+        self, db_cluster_id: str, target_state: str, check_interval: int = 30, max_attempts: int = 40
+    ) -> None:
+        """
+        Polls until the target state is reached.
+        An error is raised after a max number of attempts.
+
+        :param db_cluster_id: The ID of the target DB cluster.
+        :param target_state: Wait until this state is reached
+        :param check_interval: The amount of time in seconds to wait between attempts
+        :param max_attempts: The maximum number of attempts to be made
+
+        """
+
+        def poke():
+            return self.get_db_cluster_state(db_cluster_id)
+
+        target_state = target_state.lower()
+        self._wait_for_state(poke, target_state, check_interval, max_attempts)
+        self.log.info("DB cluster snapshot '%s' reached the '%s' state", db_cluster_id, target_state)
```

Comment:
I thinks currently we have a different method for waiting operations in new hooks?

@vincbeck @ferruzzi @vandonr-amz Am I right?

Reply:
Correct! Please use the function `waiter` defined [here](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/utils/waiter.py)

Reply:
Sorry to contradict Vincent, but we should be standardizing on [this](https://github.com/apache/airflow/tree/main/airflow/providers/amazon/aws/waiters) new Waiter implementation which offloads a lot of the work to the boto API instead.

Reply:
Nice catch! No worries at all, it is good to be contradicted :) I forgot we implemented this. Side question, should we then deprecate the `waiter` function or is there any use case not covered by the custom waiters which the `waiter` function satisfy?

Reply:
I think Syed and I were working on them in parallel without noticing it, but whether there are usecases where his (the one you linked first) is the better answer... I don't _think_ there are, but I could be mistaken.

And yeah, I guess we should flag one as deprecated, or at least leave a comment to that effect so folks don't add to the mess, and set some time aside to do the conversions.  Batch is another that has it's own [unique way of re-implementing the boto waiter](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/batch_waiters.py) and needs to get moved over to a standardized approach at some point.  

I think it would be scope creep to have any of that done here, but the new waiters should definitely be done the "right" way at the very least.